mesonbuild/wrap/wraptool.py

```diff
@@ -58,10 +59,24 @@ def add_arguments(parser: 'argparse.ArgumentParser') -> None:
     p.add_argument('project_path')
     p.set_defaults(wrap_func=promote)
 
+@lru_cache(maxsize=None)
```

Comment:
I'm not a fan of causing an http query on every configuration that encounters a missing dependency. This lookup should be done using only local data, meaning caching the release file in the file system. Alternatively it could download the file if wrap mode is not `nodownload` or something similar. Also `maxsize=None` is meaningless here since the function takes no arguments.

Reply:
`--wrap-mode=nodownload` should indeed skip this. You think we should not download it implicitely, but force user to run a command (e.g. `meson wrap update-db`) that store the DB locally and use that?

> Also `maxsize=None` is meaningless here since the function takes no arguments.

Good point, that's blind copy/pasted from elsewhere we used it.